### PR TITLE
SingleFile diagnostic support - Add export table and DotNetRuntimeInfo to dumps

### PR DESCRIPTION
**Pull request by hoyosjs**
- DACize PEDecoder::GetExport

DACize the extraction of exports and account for mapped vs non-mapped versions. @trylek, could you please check this first commit.

- Ensure export table and runtimeinfo export are always present in a minidump

### CODE REVIEW

File: src/coreclr/debug/daccess/enummem.cpp [hunk 1]
```diff
@@ -1558,6 +1558,33 @@ HRESULT ClrDataAccess::EnumMemCLRMainModuleInfo()
...
+
+        ReportMem(pExportTablesOffset, size, true);
+
+        // This following part is a bit manual. We have to do this as we generate this
+        // structure from the DAC binary itself, so we have a chicken and egg problem.
+        // Instead we break the cycle by hardcoding the size  of the struct and assert
+        // the size of the definition with a warning to update this side. Only single
+        // files will have the runtime info struct.
+        const LPCSTR RuntimeInfoExportName = "DotNetRuntimeInfo";
+        const UINT32 RuntimeInfoExportSize = 96;
```
Comment (mikem8361) [inline src/coreclr/debug/daccess/enummem.cpp:1574]: Can't we put these defines in runtimeinfo.h? I guess I don't understand this cycle.  "runtimeinfo.h" isn't generated.

File: src/coreclr/debug/daccess/enummem.cpp [hunk 2]
```diff
@@ -1558,6 +1562,28 @@ HRESULT ClrDataAccess::EnumMemCLRMainModuleInfo()
         status = E_UNEXPECTED;
     }
 
+#if defined(FEATURE_SINGLE_FILE_DIAGNOSTICS)
```
Comment (mikem8361) [inline src/coreclr/debug/daccess/enummem.cpp:1565]: Where/when is FEATURE_SINGLE_FILE_DIAGNOSTICS defined?

Comment (hoyosjs) [reply src/coreclr/debug/daccess/enummem.cpp:1565]: Uncommited file based on the feature flag in src/coreclr/clrfeatures.cmake; I removed it as it doesn't make much sense to compile without.

**Review by mikem8361 — APPROVED** [review]

**Comment by hoyosjs** [comment]
Tested this again with a single file and I see the backing memory getting enumerated.

**Comment by hoyosjs** [comment]
@trylek, had to fix a bug in the `GetExport` logic, it was using the index in the name table as an offset to the address table, but according to the spec it is an index into the ordinal table, which will give the offset for the address table. Can you PTAL?

**Review by mikem8361 — APPROVED** [review]

**Review by trylek — APPROVED** [review]
Looks good, thanks for the fix. Can you please double-check that the corresponding code in R2RDump is correct? I think it is but you have the structure in fresh memory now ;-).

https://github.com/dotnet/runtime/blob/64303750a9198a49f596bcc3aa13de804e421579/src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/PEReaderExtensions.cs#L19

**Comment by hoyosjs** [comment]
@trylek I've verified both the PEReader extensions and `ILCompiler.PEWriter.SectionBuilder.SerializeExportSection`, and both do the ordinal table indirection correctly. Are the tests in the composite pipeline expected to fail like this?

**Comment by trylek** [comment]
@hoyoys - Thank you! For the composite runs, all failures I'm seeing correspond to known issues. I have a long-haul PR https://github.com/dotnet/runtime/pull/51416 to fix part of them but it's taking a bit due to a complex discussion on the proper fix and the nullrefs on OSX arm64 are a known codegen issue unrelated to Crossgen2 I believe.

**Pull request merged**